test/common/upstream/outlier_detection_impl_test.cc

```diff
@@ -1894,6 +1926,107 @@ TEST_F(OutlierDetectorImplTest, MaxEjectTimeNotAlligned) {
+
+  EXPECT_EQ(100UL, detector->config().intervalMs());
+  EXPECT_EQ(10000UL, detector->config().baseEjectionTimeMs());
+  EXPECT_EQ(13000UL, detector->config().maxEjectionTimeJitterMs());
+}
+
+// Test verifies that detector is properly initialized with
+// default max_ejection_time_jitter when it is
+// not specified in config.
+TEST_F(OutlierDetectorImplTest, DetectorStaticConfigDefaultMaxEjectionTimeJitter) {
+  const std::string yaml = R"EOF(
+interval: 0.1s
+base_ejection_time: 10s
+  )EOF";
+  envoy::config::cluster::v3::OutlierDetection outlier_detection;
+  TestUtility::loadFromYaml(yaml, outlier_detection);
+  EXPECT_CALL(*interval_timer_, enableTimer(std::chrono::milliseconds(100), _));
+  std::shared_ptr<DetectorImpl> detector(DetectorImpl::create(
+      cluster_, outlier_detection, dispatcher_, runtime_, time_system_, event_logger_, random_));
+
+  EXPECT_EQ(100UL, detector->config().intervalMs());
+  EXPECT_EQ(10000UL, detector->config().baseEjectionTimeMs());
+  EXPECT_EQ(0UL, detector->config().maxEjectionTimeJitterMs());
+}
+
+// Test verifies that jitter is between 0s and 10s
+// when max_ejection_time_jitter is set to 10s.
+TEST_F(OutlierDetectorImplTest, EjectionTimeJitterIsInRange) {
+  // Setup max_ejection_time_jitter time to 10 secs.
+  ON_CALL(runtime_.snapshot_, getInteger(MaxEjectionTimeJitterMsRuntime, _))
+      .WillByDefault(Return(10000UL));
+  // Add host.
+  EXPECT_CALL(cluster_.prioritySet(), addMemberUpdateCb(_));
+  addHosts({"tcp://127.0.0.1:80"});
+  EXPECT_CALL(*interval_timer_, enableTimer(std::chrono::milliseconds(10000), _));
+  std::shared_ptr<DetectorImpl> detector(DetectorImpl::create(cluster_, empty_outlier_detection_,
+                                                              dispatcher_, runtime_, time_system_,
+                                                              event_logger_, random_));
+  detector->addChangedStateCb([&](HostSharedPtr host) -> void { checker_.check(host); });
+  // Set the return value of random().
+  EXPECT_CALL(random_, random()).WillOnce(Return(123456789UL));
+  // Trigger host eject.
+  EXPECT_CALL(checker_, check(hosts_[0]));
+  EXPECT_CALL(*event_logger_, logEject(std::static_pointer_cast<const HostDescription>(hosts_[0]),
+                                       _, envoy::data::cluster::v3::CONSECUTIVE_5XX, true));
+  loadRq(hosts_[0], 5, 500);
+  // Make sure that the host has been ejected.
+  EXPECT_TRUE(hosts_[0]->healthFlagGet(Host::HealthFlag::FAILED_OUTLIER_CHECK));
+  EXPECT_EQ(1UL, outlier_detection_ejections_active_.value());
+  // Get the jitter from the host monitor
+  // and make sure that it is strictly between
+  // 0 and 10000. We know that it shouldn't equal 0 or 10000
+  // since we chose the parameters.
+  auto host_monitor = detector->getHostMonitors().find(hosts_[0])->second;
+  uint64_t jitter = host_monitor->getJitter().count();
+  EXPECT_THAT(jitter, testing::AllOf(testing::Gt(0UL), testing::Lt(10000UL)));
```

Comment:
If you know what the number generator returned above (line 1983), why not expect an exact value?

Reply:
You're right, fixed!

Reply:
@adisuissa Fixed nits. One workflow was cancelled for running for too long, but it is not related to my code.